src/Sentry.Unity/SentryUnity.cs

```diff
@@ -33,75 +38,89 @@ public static void Init(Action<SentryUnityOptions> sentryUnityOptionsConfigure)
                     }
                 }
 
-                var sentryDotNet = SentrySdk.Init(options);
+                DotnetSdk = SentrySdk.Init(options);
 
-                if (options.AttachScreenshot)
+                if (Options.AttachScreenshot)
                 {
                     SentrySdk.ConfigureScope(s =>
                         s.AddAttachment(new ScreenshotAttachment(
                             new ScreenshotAttachmentContent(options, SentryMonoBehaviour.Instance))));
                 }
 
-                if (options.NativeContextWriter is { } contextWriter)
+                if (Options.NativeContextWriter is { } contextWriter)
                 {
                     SentrySdk.ConfigureScope((scope) =>
                     {
                         var task = Task.Run(() => contextWriter.Write(scope)).ContinueWith(t =>
                         {
                             if (t.Exception is not null)
                             {
-                                options.DiagnosticLogger?.LogWarning(
+                                Options.DiagnosticLogger?.LogWarning(
                                     "Failed to synchronize scope to the native SDK: {0}", t.Exception);
                             }
                         });
                     });
                 }
 
+                ApplicationAdapter.Instance.Quitting += Close;
+            }
+        }
 
-                ApplicationAdapter.Instance.Quitting += () =>
+        /// <summary>
+        /// Closes the Sentry Unity SDK
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static void Close()
+        {
+            Options?.DiagnosticLogger?.LogDebug("Closing the sentry-dotnet SDK");
+            try
+            {
+                ApplicationAdapter.Instance.Quitting -= Close;
+
+                if (Options is not null)
                 {
-                    options.DiagnosticLogger?.LogDebug("Closing the sentry-dotnet SDK");
-                    try
-                    {
-                        sentryDotNet.Dispose();
-                    }
-                    finally
-                    {
-                        try
-                        {
-                            // We don't really need to close, Windows would release the lock anyway, but let's be nice.
-                            LockFile?.Close();
-                        }
-                        catch (Exception ex)
-                        {
-                            options.DiagnosticLogger?.Log(SentryLevel.Warning,
-                                "Exception while releasing the lockfile on the config directory.", ex);
-                        }
-                    }
-                };
+                    Options.NativeSupportCloseCallback?.Invoke();
+                    Options.NativeSupportCloseCallback = null;
+                }
+
+                DotnetSdk?.Dispose();
+            }
+            finally
```

Comment:
Do we expect this to throw? If so, maybe catch & log, same as below for LockFile, instead of having a `finally` branch?